diff --git a/services/server/src/server/controllers/verification/verification.common.ts b/services/server/src/server/controllers/verification/verification.common.ts
index d9317900..e9a0b13c 100644
--- a/services/server/src/server/controllers/verification/verification.common.ts
+++ b/services/server/src/server/controllers/verification/verification.common.ts
@@ -460,7 +460,7 @@ export const verifyContractsInSession = async (
       logger.info("dryRun verification", {
         sessionId: session.id,
       });
-      return;
+      continue;
     }
     if (match.runtimeMatch || match.creationMatch) {
       await storageService.storeMatch(checkedContract, match);
